test/extensions/filters/http/cache/cache_filter_integration_test.cc

```diff
@@ -26,6 +36,51 @@ class CacheIntegrationTest : public Event::TestUsingSimulatedTime,
     codec_client_ = makeHttpConnection(makeClientConnection((lookupPort("http"))));
   }
 
+  Http::TestRequestHeaderMapImpl httpRequestHeader(std::string method, std::string authority) {
+    return {{":method", method},
+            {":path", absl::StrCat("/", protocolTestParamsToString({GetParam(), 0}))},
+            {":scheme", "http"},
+            {":authority", authority}};
+  }
+
+  Http::TestResponseHeaderMapImpl httpResponseHeadersForBody(
+      const std::string& body, const std::string& cache_control = "public,max-age=3600",
+      std::initializer_list<std::pair<std::string, std::string>> extra_headers = {}) {
+    Http::TestResponseHeaderMapImpl response = {{":status", "200"},
+                                                {"date", formatter_.now(simTime())},
+                                                {"cache-control", cache_control},
+                                                {"content-length", std::to_string(body.size())}};
+    for (auto& header : extra_headers) {
+      response.addCopy(header.first, header.second);
+    }
+    return response;
+  }
+
+  IntegrationStreamDecoderPtr sendHeaderOnlyRequestAwaitResponse(
+      const Http::TestRequestHeaderMapImpl& headers,
+      std::function<void()> simulate_upstream = []() {}) {
+    IntegrationStreamDecoderPtr response_decoder = codec_client_->makeHeaderOnlyRequest(headers);
+    simulate_upstream();
+    // Wait for the response to be read by the codec client.
+    EXPECT_TRUE(response_decoder->waitForEndStream());
+    EXPECT_TRUE(response_decoder->complete());
+    return response_decoder;
+  }
+
+  std::function<void()> simulateUpstreamResponse(const Http::TestResponseHeaderMapImpl& headers,
+                                                 const std::string* body) {
```

Comment:
nit: Try `OptRef<std::string>`, found in include/envoy/common/optref.h for a more Envoy-idiomatic way of doing this.